pkg/credentialwatcher/secretwatcher.go

```diff
@@ -0,0 +1,129 @@
+package credentialwatcher
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// STSCredentialsSuffix is the suffix applied to account.Name to create STS Secret
+	STSCredentialsSuffix = "-sre-credentials"
+	// STSCredentialsDuration Duration of STS token and Console signin URL
+	STSCredentialsDuration = 43200
+	// STSCredentialsThreshold Time before STS credentials are recreated
+	STSCredentialsThreshold = 7200
+)
+
+// SecretWatcher global var for SecretWatcher
+var SecretWatcher *secretWatcher
+
+type secretWatcher struct {
+	watchInterval time.Duration
+	client        client.Client
+}
+
+// Initialize creates a global instance of the SecretWatcher
+func Initialize(client client.Client, watchInterval time.Duration) {
+	SecretWatcher = NewSecretWatcher(client, watchInterval)
+}
+
+// NewSecretWatcher returns a new instance of the SecretWatcher interface
+func NewSecretWatcher(client client.Client, watchInterval time.Duration) *secretWatcher {
+	return &secretWatcher{
+		watchInterval: watchInterval,
+		client:        client,
+	}
+}
+
+// SecretWatcher will trigger CredentialsRotator every `scanInternal` and only stop if the operator is killed or a
+// message is sent on the stopCh
+func (s *secretWatcher) Start(log logr.Logger, stopCh <-chan struct{}) {
+	log.Info("Starting the secretWatcher")
+	for {
+		select {
+		case <-time.After(s.watchInterval):
+			log.Info("secretWatcher: scanning secrets")
+			err := s.ScanSecrets(log)
+			if err != nil {
+				log.Error(err, "secretWatcher not started, credentials wont be rotated")
+			}
+		case <-stopCh:
+			log.Info("Stopping the secretWatcher")
```

Comment:
should there be a loop breaking statement here ?

Reply:
:100: :+1: